lab_1_classify_by_unigrams/main.py

```diff
@@ -23,7 +32,20 @@ def tokenize(text: str) -> list[str] | None:
 
 
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
-    """
+    if isinstance(tokens, list) and all(isinstance(token, str) for token in tokens):
+        d = {}
```

Comment:
rename the variable.
try to avoid naming variables with one letter.

Reply:
done